### ISSUE TITLE
regex bug: new line char ^ not being interpreted

### ISSUE THREAD

**Issue by minispooner** [projectdiscovery/httpx#1000]
### httpx version:
1.2.7

### Current Behavior:
regex matches (-mr) ignore the start of line char ^ and fail to match it

### Expected Behavior:
regex matches should match a start of line with the ^ char

### Steps To Reproduce:
1. Run `echo "https://example.com" | httpx -t 1 -mr "^<html>"` in your terminal
2. It does not match anything.
3. Hit https://example.com in your browser and view the source code and you'll see line 2 does start with "<html>"

### Anything else:
1. I'm no expert in Go at all (I'm a python guy), but I think the -mr and -fe (and others?) regex input options are being incorrectly stored as a string ("") variable type and not a raw string literal (``). See https://github.com/projectdiscovery/httpx/blob/954cbe6e95a1cafadf116cd37946ae8b73135091/runner/options.go#L78 and https://github.com/projectdiscovery/httpx/blob/954cbe6e95a1cafadf116cd37946ae8b73135091/runner/options.go#L169-L170 and
https://github.com/projectdiscovery/httpx/blob/954cbe6e95a1cafadf116cd37946ae8b73135091/common/httpx/filter.go#L37
3.  See reference to raw string literals in Go here https://stackoverflow.com/questions/14183704/regular-expression-doesnt-work-in-go

**Comment by Mzack9999** [comment]
@minispooner The matching is happening on the whole response as a unique string, not line by line, therefore if the regular expression is literally `"^<html>"` it will not match the response from `example.com` as it starts with `<!doctype html>`:
```console
$ curl https://example.com
<!doctype html>
<html>
...
```

Would being able to customize the matching behavior be a helpful addition?

**Comment by minispooner** [comment]
yes, regexing line by line would be a helpful feature. I'm not sure though how processing time would be impacted at scale though, of parsing the whole file at once as compared to line by line. you could add in a quick `if regex_in_use: parse_lines() else parse_whole_response()`

**Comment by Mzack9999** [comment]
What about removing the start line condition from the regular expression (`<html>`)?

**Comment by minispooner** [comment]
that would match in this example, but you still miss out on the exactness gained from evaluating a new line ^ char

**Comment by tarunKoyalwar** [comment]
@minispooner, html response received in response is usually a single line [without new line] curl and browser shows prettyfied version of it so you can't split it into lines even if we want . and if we do it somehow it will cause more issues for use cases where matcher condition is > 1 line .

anyway not sure about your usecase but still you can do this `-mdc` matcher condition and helper functions https://nuclei.projectdiscovery.io/templating-guide/helper-functions/
